unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -136,17 +139,35 @@ private void AddFriendRequests(AddFriendRequestsPayload msg)
             TotalSentFriendRequestCount = msg.totalSentFriendRequests;
             OnTotalFriendRequestUpdated?.Invoke(TotalReceivedFriendRequestCount, TotalSentFriendRequestCount);
 
-            foreach (var userId in msg.requestedFrom)
+            List<FriendRequest> receivedFriendRequestsToAdd = new List<FriendRequest>();
+            foreach (var friendRequest in msg.requestedFrom)
             {
+                receivedFriendRequestsToAdd.Add(new FriendRequest(
+                    friendRequest.friendRequestId,
+                    friendRequest.timestamp,
+                    friendRequest.from,
+                    friendRequest.to,
+                    friendRequest.messageBody));
+
                 UpdateFriendshipStatus(new FriendshipUpdateStatusMessage
-                    {action = FriendshipAction.REQUESTED_FROM, userId = userId});
+                    {action = FriendshipAction.REQUESTED_FROM, userId = friendRequest.from});
             }
+            OnReceivedFriendRequestsAdded?.Invoke(receivedFriendRequestsToAdd);
```

Comment:
i think it would be better to emit the event after all the processing is done, what do you think?

Reply:
I don't need these events anymore as the result list is already managed by the consumer.